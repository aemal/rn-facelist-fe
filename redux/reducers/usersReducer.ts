import {
  IUsers,
  Type,
  IUsersAction
} from "../types/userTypes";

const initialState: IUsers = { usersList: [] };

export const usersReducer = (
  state = initialState,
  action: IUsersAction
): IUsers => {
  switch (action.type) {
    case Type.FETCH_USERS:
      return {
        ...state,
        usersList: action.payload
      };
    default:
      return {
        ...state,
      }
  }
}
