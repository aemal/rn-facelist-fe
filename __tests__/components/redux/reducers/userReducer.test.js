import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk';
import {usersReducer} from '../../../../redux/reducers/usersReducer'
const middleware = [thunk];
const mockStore = configureMockStore(middleware);

//mock data
const initialState = {
  users: {
    usersList: [
      { "id": "1", "avatar": "https://i.pravatar.cc/150?img=1", "name": "Morgan James" },
      { "id": "2", "avatar": "https://i.pravatar.cc/150?img=2", "name": "Esme Coleman" },
    ]
  },
};

//store initialization
let store = mockStore(initialState);

//clear actions
beforeEach(() => {
  store.clearActions();
});

//test case for userReducer
it('UserReducer Test', () => {
  const action = {type: "FETCH_USERS", payload: initialState.users.usersList};
  expect(usersReducer(undefined,action)).toMatchSnapshot();
});