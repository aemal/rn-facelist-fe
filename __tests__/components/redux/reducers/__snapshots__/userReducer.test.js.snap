// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserReducer Test 1`] = `
Object {
  "usersList": Array [
    Object {
      "avatar": "https://i.pravatar.cc/150?img=1",
      "id": "1",
      "name": "Morgan James",
    },
    Object {
      "avatar": "https://i.pravatar.cc/150?img=2",
      "id": "2",
      "name": "Esme Coleman",
    },
  ],
}
`;
